data/data-pipeline/data_pipeline/tests/sources/test_etl_base.py

```diff
@@ -0,0 +1,549 @@
+# pylint: disable=protected-access,unsubscriptable-object
+import copy
+import os
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from data_pipeline.config import settings
+from data_pipeline.etl.base import ExtractTransformLoad, ValidGeoLevel
+from data_pipeline.tests.conftest import copy_data_files
+from data_pipeline.tests.sources.example.etl import ExampleETL
+from data_pipeline.utils import get_module_logger
+
+logger = get_module_logger(__name__)
+
+
+class TestETL:
+    """A base class that can be inherited by all other ETL tests.
+
+    Note: every method that does *not* need to be reimplemented by child classes has
+    the test name pattern of `test_*_base`. All other tests need to be reimplemented.
+    """
+
+    # In every child test class, change this to the class of the ETL being tested.
+    _ETL_CLASS = ExampleETL
+
+    # In every child test class, change this to the directory of the text fixtures.
+    _DATA_DIRECTORY_FOR_TEST = (
```

Comment:
Looks like we can't use `__init__` (https://stackoverflow.com/questions/62460557/cannot-collect-test-class-testmain-because-it-has-a-init-constructor-from), but I implemented a different version of this. Good idea!